tests/data/test_types.py

```diff
@@ -146,6 +140,52 @@ def test_valid_trajectories(
         for traj in trajs:
             assert len(traj) == length
 
+    def test_equals(
```

Comment:
This test turned out beautiful!
In the spirit of only testing one thing in each test, we could split it up into multiple tests.
I would propose tests called: `test_that_traj_unequal_to_primitives`, `test_that_traj_equal_to_self`, `test_that_traj_equal_to_copy`, `test_that_traj_unequal_to_copy_with_reward`, `test_that_traj_unequal_to_perturbed`.
But I can do this later when I do a general sweep through all of the tests.

Reply:
Thanks for the suggestion. I've split up the tests. I went for a slightly different decomposition to avoid having some tiny tests with lots of boilerplate.